src/interpreter.c

```diff
@@ -14,34 +14,27 @@
 extern "C" {
 #endif
 
-static jl_value_t *eval(jl_value_t *e, jl_value_t **locals, size_t nl, size_t ngensym);
-static jl_value_t *eval_body(jl_array_t *stmts, jl_value_t **locals, size_t nl, size_t ngensym,
+static jl_value_t *eval(jl_value_t *e, jl_value_t **locals, jl_lambda_info_t *lam);
+static jl_value_t *eval_body(jl_array_t *stmts, jl_value_t **locals, jl_lambda_info_t *lam,
                              int start, int toplevel);
 jl_value_t *jl_eval_module_expr(jl_expr_t *ex);
 int jl_is_toplevel_only_expr(jl_value_t *e);
 
 jl_value_t *jl_interpret_toplevel_expr(jl_value_t *e)
 {
-    return eval(e, NULL, 0, 0);
+    return eval(e, NULL, NULL);
 }
 
-JL_DLLEXPORT jl_value_t *jl_interpret_toplevel_expr_in(jl_module_t *m,
-                                                       jl_value_t *e,
-                                                       jl_svec_t *local_syms,
-                                                       jl_svec_t *local_vals)
+JL_DLLEXPORT jl_value_t *jl_interpret_toplevel_expr_in(jl_module_t *m, jl_value_t *e,
+                                                       jl_lambda_info_t *lam)
```

Comment:
this seems to assume that `local_vals == lam->sparam_vals`, but that is false for the LambdaInfo reached at `LambdaInfo->unspecialized` and `LambdaInfo`)